src/Voron/Data/CompactTrees/CompactKey.cs

```diff
@@ -19,20 +19,30 @@ namespace Voron.Data.CompactTrees;
 public sealed unsafe class CompactKey : IDisposable
 {
     public static readonly CompactKey NullInstance = new();
-    
-    private static readonly ArrayPool<byte> StoragePool = ArrayPool<byte>.Shared;
-    private static readonly ArrayPool<long> KeyMappingPool = ArrayPool<long>.Shared;
+
+    [ThreadStatic]
+    private static ArrayPool<byte> StoragePool;
```

Comment:
Is there a scenario where we need _multiple_ buffers? shouldn't this just be a buffer?
In general, I would rather have a per core instead of per thread.

Otherwise, we may end up with a thread that holds a lot of buffer in the pool that aren't used.

Reply:
Yes, we could use one per core. I will address that.

Reply:
Done